src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java

```diff
@@ -688,10 +688,9 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
 
   @Override
   @Transactional
-  public Mono<BookingTO> updateBookingByReferenceCarrierBookingRequestReference(
-      String carrierBookingRequestReference, BookingTO bookingRequest) {
-
-    if (!carrierBookingRequestReference.equals(
+  public Mono<BookingResponseTO> updateBookingByReferenceCarrierBookingRequestReference(String carrierBookingRequestReference, BookingTO bookingRequest) {
```

Comment:
I think we need some checks similar to what we have at the beginning of `createBooking` concerning `getIsImportLicenseRequired`, `getImportLicenseReference`, `getIsExportDeclarationRequired`, `getExportDeclarationReference`.

Reply:
Yes, will include that.